tools/preflight/preflight.go

```diff
@@ -836,3 +856,26 @@ func (o *Run) createRestorePodFromSnapshot(ctx context.Context, volSnapshot *uns
 
 	return restorePod, nil
 }
+
+// function to convert Unstructured Object to YAML
+func toYAML(obj *unstructured.Unstructured) ([]byte, error) {
+	jsonOut, mErr := obj.MarshalJSON()
+	if mErr != nil {
+		return nil, fmt.Errorf("error marshaling object :: %s", mErr.Error())
+	}
+	yamlOut, jErr := yaml.JSONToYAML(jsonOut)
+	if jErr != nil {
+		return nil, fmt.Errorf("error converting json object to yaml :: %s", jErr.Error())
+	}
+	return yamlOut, nil
+}
+
+// function to convert Structured Object to YAML
+func objToYAML(o interface{}) []byte {
+	objYAML, yErr := yaml.Marshal(o)
+	if yErr != nil {
+		fmt.Printf("error converting strucuted object to yaml :: %s", yErr.Error())
+		return nil
+	}
+	return objYAML
+}
```

Comment:
func `objToYAML` can be used for both structured and unstructured types.
Return error along with byte data.